[craigtonlian] iP

craigtonlian opened the pull request:
# GrumpyGordon
> "I'm Gordon Ramsay, for goodness sake: people *know* I'm volatile." - Gordon Ramsay ([source](https://www.brainyquote.com/quotes/gordon_ramsay_455540))

GrumpyGordon helps you get shit done. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/craigtonlian/ip/releases/download/A-Jar/grumpygordon.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :blush:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

___

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

craigtonlian committed:
added Level-0: Rename, Greet, Exit

craigtonlian committed:
added Level-1: Echo

craigtonlian committed:
added Level-2: Add, List

craigtonlian committed:
added Level-3: Mark as Done

craigtonlian committed:
added Level-4: ToDos, Events, Deadlines

craigtonlian committed:
added A-TextUiTesting: Automated Text UI Testing

craigtonlian committed:
added Level-5: Handle Errors

craigtonlian committed:
added Level-6: Delete

craigtonlian committed:
added A-Enums: Use Enums

craigtonlian committed:
added A-Enums: Use Enums

craigtonlian committed:
Merge branch 'master' of https://github.com/craigtonlian/ip

# Conflicts:
#	src/main/java/Command.java

craigtonlian committed:
added: javadocs comments for classes

craigtonlian committed:
Add Level-7: Save

craigtonlian committed:
Merge branch 'branch-Level-7'

craigtonlian committed:
Add Level-8: Dates and Times

craigtonlian committed:
Merge branch 'branch-Level-8'

craigtonlian committed:
Add A-MoreOOP: Use More OOP

craigtonlian committed:
Merge branch 'branch-A-MoreOOP'

craigtonlian committed:
Add A-Packages: Organize into Packages

craigtonlian committed:
Merge branch 'branch-A-packages'

craigtonlian committed:
Merge branch 'add-gradle-support'

craigtonlian committed:
Add A-JUnit: Add JUnit Tests

craigtonlian committed:
Merge branch 'A-JUnit'

craigtonlian committed:
Add A-Jar: Create a JAR File

craigtonlian committed:
Merge branch 'branch-A-Jar'

craigtonlian committed:
Add JavaDoc comments

craigtonlian committed:
Add coding standard edits

craigtonlian committed:
Add Level-9: Find

craigtonlian committed:
Merge branch 'branch-Level-9'

craigtonlian committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/grumpygordon/ui/Ui.java

craigtonlian committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/grumpygordon/parser/Parser.java
#	src/main/java/grumpygordon/storage/Storage.java

craigtonlian committed:
Fix runtest.bat error when run with existing task list storage file

craigtonlian committed:
Merge branch 'branch-textui-fixes'

craigtonlian committed:
Add A-CheckStyle: Use CheckStyle

craigtonlian committed:
Merge branch 'branch-A-CheckStyle'

craigtonlian committed:
Add gradle dependencies

craigtonlian committed:
Add launcher integration with javafx

craigtonlian committed:
Add ScrollPane, Container, TextField, SendButton ui

craigtonlian committed:
Add DialogBox

craigtonlian committed:
Add DialogBox custom behaviors

tohpinren commented on src/main/java/grumpygordon/GrumpyGordon.java:
```diff
@@ -0,0 +1,61 @@
+
+/**
+ * GrumpyGordon Chatbot
+ */
```
Nice. I like that you added javadocs already

tohpinren commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,269 @@
+    /**
+     * Regex for datetime.
+     */
+    private static final String DATETIME_REGEX = "\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]) \\d{2}:\\d{2}";
```
I like your naming of constants

tohpinren commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,269 @@
+
+        try {
+            switch (type) {
+            case "T":
```
Good that there is no indentation for case clauses.

tohpinren commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,269 @@
+            if (!Pattern.matches(MARK_REGEX, userInput)) {
+                throw new GrumpyGordonInvalidCommandException("Command syntax for mark is incorrect.\n");
+            }
+
```
I like how there is whitespace to make the code blocks more readable

tohpinren commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,269 @@
+                return new Deadline(description, parseDateTime(deadlineBy), isDone);
+            case "E":
+                // Parse the saved format for Event
+                // Example: "E | 0 | Team meeting | 2023-09-01 12:00 | 2023-09-02 14:00"
```
I like the additional comments to make it clearer

tohpinren commented on src/main/java/grumpygordon/tasks/Task.java:
```diff
@@ -0,0 +1,74 @@
+     * Returns the Task from the save format.
+     * @return Task generated from the save format
+     */
+    public static Task fromSaveFormat(String line) throws GrumpyGordonException {
```
Should the name of this method be a verb?

tohpinren commented on src/main/java/grumpygordon/tasks/Todo.java:
```diff
@@ -0,0 +1,34 @@
+     * @return String representation of the save format of a Todo task
+     */
+    @Override
+    public String toSaveFormat() {
```
Should the name of this method be a verb?

tohpinren reviewed (commented):
Overall, your code is very readable and follows almost all the coding standards given. Your javadocs are clearly written and there are additional comments for confusing parts too. Good job!

craigtonlian committed:
Add gui without response loop

craigtonlian committed:
Integrate response loop with GUI

craigtonlian committed:
Merge branch 'branch-Level-10'

devanshubisht commented on src/main/java/grumpygordon/GrumpyGordon.java:
```diff
@@ -0,0 +1,65 @@
+        } catch (GrumpyGordonInitialisationException e) {
+            this.tasks = new TaskList();
+        }
+        this.ui = new Ui(tasks, storage);
```
Perhaps you can remove your UI class as it is not used anymore due JavaFx.

devanshubisht commented on src/main/java/grumpygordon/parser/Parser.java:
Extremely clean and well written logic and code.

devanshubisht commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,304 @@
+        String[] args;
+
+        switch (command) {
+        case "bye":
```
Perhaps you could use Enums for the different cases.

devanshubisht commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,304 @@
+    /**
+     * Regex for find command.
+     */
+    private static final String FIND_REGEX = "find\\s.*$";
+
+    /**
+     * Regex for deadline command.
+     */
+    private static final String DEADLINE_INFO_REGEX = ".*\\s+/by\\s+.*";
+    /**
+     * Regex for event command.
+     */
+    private static final String EVENT_INFO_REGEX = ".*\\s+/from\\s+.*\\s+/to\\s+.*";
+    /**
+     * Regex for datetime.
+     */
+    private static final String DATETIME_REGEX = "\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01]) \\d{2}:\\d{2}";
```
I like how you declared this here, making it easier for you to parse the inputString later.

devanshubisht commented on src/main/java/grumpygordon/parser/Parser.java:
```diff
@@ -0,0 +1,304 @@
+                throw new GrumpyGordonInvalidCommandException(e.getMessage());
+            }
+        default:
+            break;
```
Perhaps you can consider putting the InvalidCommanException here instead

devanshubisht reviewed (commented):
Overall, this is very well-written code. It is easy to understand and you have complied well with the coding standards and quality. Perhaps you can begin to remove the methods you are no longer using due to the addition of JavaFx

craigtonlian committed:
Add assertions to Parser class

No assertions used in Parser class.

Assertions help to ensure expected behaviour.

Let's update Parser class with the necessary assertions.

craigtonlian committed:
Unify Ui and Launcher with GrumpyGordon class

Ui class and Launcher class is redundant.

With the integration of JavaFX, there is no further need for a separate
Ui class to encapsulate the interface components.

Let's refactor away the unnecessary code portions for better
maintainability.

craigtonlian committed:
Merge pull request #1 from craigtonlian/branch-A-CodeQuality

Merge A-CodeQuality to master

craigtonlian committed:
Merge branch 'master' of https://github.com/craigtonlian/ip

craigtonlian committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/grumpygordon/ui/Ui.java

craigtonlian committed:
Merge pull request #2 from craigtonlian/branch-A-Assertions

Merge A-Assertions

craigtonlian committed:
Set up CI

No CI is set up with github workflow.

CI will improve automated build processes when certain project events
are triggered.

craigtonlian committed:
Add task sorting function

Tasks are sorted by order of insertion.

Tasks can be more easily located if sorted alphabetically.

Let's implement the functionality to allow tasks to sorted
alphabetically.

craigtonlian committed:
Merge branch 'branch-C-Sort'

craigtonlian committed:
Merge branch 'master' into branch-A-CI

craigtonlian committed:
Merge pull request #3 from craigtonlian/branch-A-CI

Merge branch-A-CI

craigtonlian committed:
Improve GUI with padding and colors

craigtonlian committed:
Merge branch 'branch-A-BetterGui'

craigtonlian committed:
Update user guide readme

craigtonlian committed:
Merge branch 'branch-A-UserGuide'

craigtonlian committed:
Fix checkstyle error in Main

craigtonlian committed:
Merge branch 'branch-A-UserGuide'

craigtonlian committed:
Fix formatting of readme

craigtonlian committed:
Merge branch 'branch-A-UserGuide'

craigtonlian committed:
Update javadocs

craigtonlian committed:
Merge branch 'branch-A-UserGuide'

craigtonlian committed:
Refactor long methods

craigtonlian committed:
Merge branch 'branch-refactor'

craigtonlian committed:
Add v0.2 jar file download link to User Guide

craigtonlian committed:
Merge branch 'branch-refactor'

craigtonlian committed:
Add testcases and javadocs

craigtonlian committed:
Merge branch 'branch-fixes'

craigtonlian committed:
Update readme with link to v0.3 jar file

craigtonlian committed:
Merge branch 'branch-update-readme'

craigtonlian committed:
Fix checkstyle errors

craigtonlian committed:
Merge branch 'branch-fix-checkstyle'
